### DISCUSSION TITLE
Contraction time

### DISCUSSION THREAD

**Discussion by fmozafari** [tencent-quantum-lab/tensorcircuit#195] [Q&A]
Hello, I'm running circuittensor for benchmarks such as GHZ, I could get the results regarding path finder but could you please let me know how can I get the contraction execution time? It was mot clear for me how can I do that. Thanks!

**Comment by refraction-ray** [comment]
by execution time, you mean theoretical flops or actual running time? if it is the latter, you can just run the function with circuit evaluation (say `c.expectation()`) and profile the time for such function? The only subtlety is when the function is jitted, the first run will takes a longer time and you have to profile the running time from the second run of the function

**Comment by fmozafari** [comment]
Thanks for the answer! So, just putting time.time() before and after the function is enough to get the time? and is it related to CPU or GPU?

**Reply by refraction-ray** [reply]
> just putting time.time() before and after the function is enough to get the time

yes, the bulk time will be the contraction running time

>  is it related to CPU or GPU?

of course, the running time on different hardware can be very different
